pkg/k8shandler/deployment.go

```diff
@@ -554,48 +563,63 @@ func (node *deploymentNode) isChanged() bool {
 		nodeContainer := node.self.Spec.Template.Spec.Containers[index]
 		desiredContainer := desired.Spec.Template.Spec.Containers[index]
 
-		if nodeContainer.Resources.Requests == nil {
-			nodeContainer.Resources.Requests = v1.ResourceList{}
-		}
+		var updatedContainer v1.Container
+		updatedContainer, changed = updateResources(node, nodeContainer, desiredContainer)
```

Comment:
If changed was set to `true` by an earlier loop iteration, this may reset it to `false`, will will cause `isChanged` to return the wrong result code.  maybe something like this:
```go
                var updatedContainer v1.Container
                var resourceChanged bool
 		if updatedContainer, resourceChanged = updateResources(node, nodeContainer, desiredContainer); resourceChanged {
                        changed = true
                }
```